src/widgets/panel/panel.tsx

```diff
@@ -248,7 +256,13 @@ const panel: PanelWidget = widgetParams => {
         return undefined;
       },
       render(...args) {
-        const [options] = args;
+        const [renderOptions] = args;
+
+        const options = {
+          ...renderOptions,
+          widgetRenderState: (widget.getWidgetRenderState?.(renderOptions) ||
+            {}) as any,
```

Comment:
Hmmm here 

```
widget.getWidgetRenderState?.(renderOptions) || {}
```

The type of it is `unknown`, because literally we don't know it there. So without casting, it cannot be passed to `hidden` and `collapsed`:

```js
hidden: Boolean(hidden(options)),
collapsed: Boolean(collapsedFn(options)),
```

But once we cast and pass it there, later on the user-land, as long as they provide the type to `panel()`, they will get benefit.
What do you think? I feel like I've come to a dead-end.

Reply:
I made this pr to help: https://github.com/algolia/instantsearch.js/pull/4558